baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -0,0 +1,110 @@
+"""Default preset for Gaussian process surrogates."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from attrs import define
+
+from baybe.kernels.basic import MaternKernel, ScaleKernel
+from baybe.kernels.priors.basic import GammaPrior
+from baybe.surrogates.gaussian_process.kernel_factory import KernelFactory
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+    from baybe.kernels.base import Kernel
+    from baybe.searchspace.core import SearchSpace
+
+
+@define
+class DefaultKernelFactory(KernelFactory):
+    """A factory providing the default kernel for the Gaussian process surrogate.
+
+    The logic is adapted from EDBO (Experimental Design via Bayesian Optimization).
+
+    References:
+    *   https://github.com/b-shields/edbo
+    *   https://doi.org/10.1038/s41586-021-03213-y
+    """
+
+    def __call__(  # noqa: D102
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        # See base class.
+
+        mordred = (searchspace.contains_mordred or searchspace.contains_rdkit) and (
```

Comment:
Probably we've talked about this long time ago but: why is "mordred" only active for these two encodings? Why not for MorganFP? Since this PR finally cleans up the kernel logic, I just want to be 100% sure that we exactly replicate the EDBO logic here.

Also: I don't like `mordred` as a variable name as it doesn't make any sense when looking at the involved conditions. Can you suggest something more reasonable? Perhaps `uses_descriptors` or similar? Asking the chem-guy among us :D

Reply:
agree, the varname is a misnomer prob stemming from the fact that is was copied from edbo, but edbo has no rdkit, feel free to change

for the original edbo logic see https://github.com/b-shields/edbo/blob/9b41eac3f6d9e520547702fd5b0c7ef6441625a4/edbo/bro.py#L668

Reply:
Yes, I even double-checked the EDBO code once more when refactoring, to make sure everything is still intact. But what is the common property of `mordred` and `rdkit` that we try to express with the variable name (and which `MorgangFP` does not possess!?)

Reply:
the only thing coming to my mind is that fingerprints are all integers in {0,1} but unclear how that would affect length scale.

Other than that theres no difference, I would also suspect that the number of columns is on the same order of magnitude - 1024 features calculated for FP, mordred does 1500, rdkit ~600 (all before de-correlation)

Reply:
Ok, then I'd say we should clearly review the logic of our `DEFAULT` preset soon – it's not really great to have some default values where none of us can really justify the logic behind 😄 ... We can do so in the PR where we do the actual benchmarking of the presets, I'd suggest.

For now, let me then simply change the variable name. Given what you wrote, I have no good idea for a name though. Would `uses_descriptors` still make sense? That's the best I could come up with, based on the fact that we call the embeddings for `mordred` and `rdkit` "descriptors" in our docstrings but we call the embedding for `morganfp` "fingerprints".

Reply:
yes the fingerprints can also be called `descriptors` so feel free to change the name
also yes to refining the prior preset in another PR

Reply:
Hm ok, not what I have hoped for (because I thought that would make the distinction), but nevermind: I'll simply rename it to `uses_descriptors` and that's already better than `mordred`. Can revisit once we improve the `BAYBE` preset logic

Reply:
changed